src/server/api/routers/smart-home/entity-state.ts

```diff
@@ -0,0 +1,53 @@
+import { TRPCError } from '@trpc/server';
+
+import { ZodError, z } from 'zod';
+
+
+import { createTRPCRouter, protectedProcedure } from '../../trpc';
+
+import { findAppProperty } from '~/tools/client/app-properties';
+import { getConfig } from '~/tools/config/getConfig';
+import { HomeAssistantSingleton } from '~/tools/singleton/HomeAssistantSingleton';
+
+export const smartHomeEntityStateRouter = createTRPCRouter({
+  retrieveStatus: protectedProcedure
+    .input(
+      z.object({
+        configName: z.string(),
+        entityId: z.string().regex(/^[A-z0-9-_\.]+$/)
+      })
+    )
+    .query(async ({ input }) => {
+      const config = getConfig(input.configName);
+
+      const instances = config.apps.filter((app) => app.integration?.type == 'homeAssistant');
+
+      for (var instance of instances) {
+        const url = new URL(instance.url);
+        const client = HomeAssistantSingleton.getOrSet(url, findAppProperty(instance, 'apiKey'));
+        const state = await client.getEntityState(input.entityId);
+        
+        if (!state.success) {
+          if (!(state.error instanceof ZodError)) {
+            continue;
+          }
+          // Consola.error('Unable to handle entity state: ', state.error);
+          throw new TRPCError({
+            code: 'NOT_IMPLEMENTED',
+            message: `Unable to handle entity state. Check log for details`
```

Comment:
Error is not descriptive enough, please consider using something like [Widget: {widgetName}] at the start of the error or be more verbose and descriptive